src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -0,0 +1,488 @@
+
+            _logger.Info("Broker Account Name: " + loginHint);
+
+            var result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                BrokerConstants.AuthtokenType, null, getAccountsBundle, null,
+                null, GetPreferredLooper(callerActivity));
+
+            if (result == null)
+            {
+                _logger.Info("Android account manager AddAccount didn't return any accounts. ");
+                throw new MsalException(MsalError.NoBrokerAccountFound, "Please add the selected account to the broker");
+            }
+
+            Bundle bundleResult = (Bundle)result?.Result;
+            var accounts = bundleResult.GetString(BrokerConstants.BrokerAccounts);
+
+            if (!string.IsNullOrEmpty(accounts))
+            {
+                dynamic authResult = JArray.Parse(accounts);
+
+                foreach (JObject account in authResult)
+                {
+                    var accountData = account[BrokerResponseConst.Account];
+                    if ((accountData[BrokerResponseConst.UserName]).ToString() == loginHint)
+                    {
+                        brokerPayload.Add(BrokerParameter.HomeAccountId, accountData[BrokerResponseConst.HomeAccountId].ToString());
+                        brokerPayload.Add(BrokerParameter.LocalAccountId, accountData[BrokerResponseConst.LocalAccountId].ToString());
+                        var acc = account;
+                        return;
+                    }
+                }
+            }
+        }
+
+        public bool SayHelloToBroker(Activity callerActivity)
+        {
+            try
+            {
+                IAccountManagerFuture result = null;
+                // Callback is not passed since it is making a blocking call to get
+                // intent. Activity needs to be launched from calling app
+                // to get the calling app's metadata if needed at BrokerActivity.
+
+                Bundle HelloBundle = new Bundle();
+                HelloBundle.PutString(BrokerConstants.ClientAdvertisedMaximumBPVersionKey, BrokerConstants.BrokerProtocalVersionCode);
+                HelloBundle.PutString(BrokerConstants.ClientConfiguredMinimumBPVersionKey, "2.0");
+                HelloBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, "HELLO");
+
+                result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                    BrokerConstants.AuthtokenType, null, HelloBundle, null,
+                    null, GetPreferredLooper(callerActivity));
+
+                if (result == null)
+                {
+                    _logger.Info("Android account manager AddAccount didn't return any results. ");
+                }
+
+                Bundle bundleResult = (Bundle)result?.Result;
+
+                var bpKey = bundleResult.GetString(BrokerConstants.NegotiatedBPVersionKey);
```

Comment:
null check needed on all the bundleResults....can that result check be moved to a separate method or something?